### DISCUSSION TITLE
Make guests AI find the First Aid Room more frequently than sitting on a bench

### DISCUSSION THREAD

**Discussion by Wolfscopez** [OpenRCT2/OpenRCT2#16971] [Feature requests]
I have the suggestion of potentially making it that guests have a certain proximity to them whether they would consider going to a First Aid Room to cure their sickness rather than immediately try to find a bench nearby.

**Comment by Ryder17z** [comment]
I don't think I have ever seen people use the first aid room. Sounds like a
good idea.

**Comment by Wolfscopez** [comment]
They do, they just rarely do it, in order for them to actually use them, you pretty much have to force them to use them via placing a First Aid room directly outside of an entrance so that they walk right towards the First-Aid room. If not then they rarely ever use them.

**Comment by duncanspumpkin** [comment]
According to the peep logic they use the first aid room if they want to go on a ride and are sick. There are a bunch of other cases that can occur before wanting to go on a ride so first aid is probably a bit too low in the priority order. It wouldn't be hard to change it so that they have a look for a first aid room when sick (and not part of the find ride code).

**Comment by Wolfscopez** [comment]
I personally would like that to be implemented as I really loathe having to have First-Aid rooms pretty much outside any Roller-Coaster or nauseating ride, or having a handyman patrol a footpath making sure there is no vomit.

**Comment by IntelOrca** [comment]
I can't remember if I improved this or not already due to peeps not using their park maps correctly.

**Comment by Wolfscopez** [comment]
I think a good idea is to add to the peep logic that guests will try and find a First-Aid Room if it is in a reasonable range, if it's like on the other side of the park then for them to just sit down instead as they'd probably vomit on the way to the First-Aid room.

So sort of to add a max range as to how far away it can be before they just sit down?
I don't know if that's possible, just an idea

**Comment by Krutonium** [comment]
So basically check for a First Aid if it is within X tiles... Seems reasonable.

**Comment by Wolfscopez** [comment]
Exactly, this is to stop guests trying to walk halfway across the park just to get to a First-Aid Room, because then if that happened, they'd just vomit on the way there.

**Comment by Wolfscopez** [comment]
I don't know if you have addressed the issue or not but you also might want to go and make it higher priority for sick peeps to go and seek a first aid room over sitting on a bench if there is one within a few blocks, some sick peeps are just sitting on a bench adjacent to a first aid room

**Comment by duncanspumpkin** [comment]
@Wolfscopez the part about checking a max range is how it works at present (I think). The issue is that its too far down the peep logic and sick peeps do the sitting check earlier on.

**Comment by tellovishous** [comment]
Pardon the bump here, but yeah I agree that peeps ought to be able to use the first aid stations more often, they're seldom used as is even though they could definitely be useful. I would also add that it could be more frequent if peeps also have maps on them, it's pretty rare that I see peeps "Heading To First Aid Room" in the guest list window.

**Comment by ocalhoun6** [comment]
Yeah ... just have to remember that sick peeps walk very slowly, and having to walk for very long is likely to make them puke. So they should probably only go to the first aid stall if it's very close by. More than 5 tiles or so, and it probably makes more sense for them to just sit on a bench.

Maybe we could actually run some experiments to find out what the ideal radius is? Set up a test scenario with a nausea-inducing ride, peeps who want to use the first aid stall, and then different versions with the first aid stall progressively further away. And also one that just has plenty of benches and no first aid stall at all. Then see which versions of that test scenario produce the most/least vomit. The ones with the first aid stall very close would probably have the least vomit, gradually increasing with distance. The important thing to find out is: At what distance does heading to the first aid stall produce _more_ vomit than the control that uses benches? Once you know that, the threshold for a peep to seek the first aid stall should probably be just below that.

But yeah... I definitely do think it's kind of dumb how you can have extremely sick guests walking right past a first aid stall that could instantly cure them, or sitting on a bench right next to a first aid stall without going in.

**Comment by redace0001** [comment]
Hello !

Sorry to revive such an old topic but this feature could be really useful.
I've been putting first aid stalls for a long time without seeing them used by guests.

This forces me to put an incredible amount of benches that will all end up being used anyway given the fact that guest need a long time sat on them to feel better (so other nauseous guest will pass every single used benches before puking somewhere at some point).

I've seen some duplicate threads suggesting making the check for first aid higher in the hierarchy of peeps logic (before "checking for a bench to sit on when sick" logic).

The modification doesn't seem that complicated according to the same thread ([here](https://github.com/OpenRCT2/OpenRCT2/issues/5273#issuecomment-284165704) for example)

I haven't tried to implement it myself for now tho ...

I might add another interesting (and funny in my opinion) addition : It could be that guest that are heading to first aid stalls are not able to puke and/or have a greater speed than normal regardless of their energy levels (the funny trope of running to the closest first aid station to not vomit on your shoes when exiting a ride).

But that a little more work and I don't know if that would fit as a feature in ORCT2 or simply a plugin.
